pkg/catalog/service.go

```diff
@@ -5,16 +5,26 @@ import (
 	"github.com/openservicemesh/osm/pkg/service"
 )
 
-// GetServiceForServiceAccount returns a service corresponding to a service account
-func (mc *MeshCatalog) GetServiceForServiceAccount(sa service.K8sServiceAccount) (service.MeshService, error) {
+// GetServicesForServiceAccount returns a list of services corresponding to a service account
+func (mc *MeshCatalog) GetServicesForServiceAccount(sa service.K8sServiceAccount) ([]service.MeshService, error) {
+	services := []service.MeshService{}
 	for _, provider := range mc.endpointsProviders {
 		// TODO (#88) : remove this provider check once we have figured out the service account story for azure vms
 		if provider.GetID() != constants.AzureProviderName {
 			log.Trace().Msgf("[%s] Looking for Services for Name=%s", provider.GetID(), sa)
 			service, err := provider.GetServiceForServiceAccount(sa)
```

Comment:
will refactor provider functions (if needed) when Shashank merges his change

Reply:
Thank you!